main.go

```diff
@@ -206,15 +209,16 @@ func main() {
 		channel = loader.DefaultChannelName
 	}
 
-	if err = controllers.NewConfigReconciler(mgr).
+	configReconcileChan := make(chan event.GenericEvent, 1)
```

Comment:
One concurrent reconciliation loop sounds just sufficient. This event generation will be rare and the other controller's loop is relatively fast, so I don't think there will be that many queued items.
Actually, queuing the events doesn't make sense as a single reconciliation run of StorageOSCluster controller will go through all the configs. And going through them twice or more times will not have any effect.
Rather than extending the queue for a random X items and/or blocking on the full queue, what if we just discard (+maybe log) the event if the single-item-queue is full?